pokemongo_bot/step_walker.py

```diff
@@ -19,17 +19,17 @@ def __init__(self, bot, dest_lat, dest_lng):
             dest_lat,
             dest_lng
         )
-
-        self.speed = self.bot.config.walk_max - random() * (self.bot.config.walk_max - self.bot.config.walk_min)
+            
+        self.speed = uniform(self.bot.config.walk_min, self.bot.config.walk_max)
 
         self.destLat = dest_lat
         self.destLng = dest_lng
         self.totalDist = max(1, self.dist)
 
         if self.speed == 0:
-            self.steps = 1
-        else:
-            self.steps = (self.dist + 0.0) / (self.speed + 0.0)
+            self.speed = uniform(3,5)
```

Comment:
As I understand it, he wants to remove the teleport thing. Currently it teleports when the speed (so walk_min/max) is equal to 0.
I’m suggesting to remove this as well (so no more ```self.steps = 1```) and add the possibility to set walk_min/max to 0. This way you can force your bot to stay still (between 6 lured forts for example) and catch pokemons (currently catching pokemon would make you move).
Also having a safe default choice is a must have. With the 2 lines I’ve shown you, you got it as well (ofc you have to place them before computing the speed).